src/Microsoft.Azure.SignalRService/ServiceManagerBuilder.cs

```diff
@@ -2,25 +2,41 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System;
-using System.Collections.Generic;
+using System.IO;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
 
 namespace Microsoft.Azure.SignalRService
 {
     public class ServiceManagerBuilder : IServiceManagerBuilder
     {
+        private ServiceCollection _serviceCollection;
+
+        public ServiceManagerBuilder()
+        {
+            _serviceCollection = new ServiceCollection();
+            _serviceCollection.AddOptions();
+            _serviceCollection.AddSingleton<IConfigureOptions<ServiceManagerOptions>, ServiceManagerOptionsSetup>();
+        }
+
         public ServiceManagerBuilder WithCredentials(string connectionString)
```

Comment:
>connectionString [](start = 60, length = 16)

check null/invalid input for public methods

Reply:
Plan to remove this interface since connection can be set via with `WithOptions(...)`. Check required options in `Build()`.